pkg/server/options/options.go

```diff
@@ -181,6 +182,12 @@ func (o *Options) Complete() (*CompletedOptions, error) {
 		}
 		o.Extra.RootDirectory = filepath.Join(pwd, o.Extra.RootDirectory)
 	}
+
+	// Create the configuration root correctly before other components get a chance.
+	if err := cmd.MkdirRoot(o.Extra.RootDirectory); err != nil {
+		return nil, err
+	}
+
```

Comment:
Is there a technical reason to have this side-effect in a completion func? I would expect this to happen in the Run code path.

Reply:
Yeh it's surprising, but the `.kcp` directory gets created as a side-effect in `o.Controllers.Complete()`

Reply:
Why don't we fix that code then? I see `if err := keyutil.WriteKey(c.SAController.ServiceAccountKeyFile, encoded); err != nil` there. `WriteKey` is only mkdir+write. Just replace that with a call to a `0700` mkdir and `ioutil.WriteFile`.

Reply:
We could do that, but that still results in the root config dir being created as a side effect. It's better to create it intentionally up front rather than depending on this spooky action at a distance.

Reply:
I can see that if we move MkdirRoot into pkg/server and make it private. Then I would also be fine with the logging output. Am not a fan of helpers that are not used and probably will never be used by other code.

Reply:
Sounds good!